src/main/java/seedu/contax/logic/parser/RangeCommandParser.java

```diff
@@ -27,18 +29,26 @@ public RangeCommand parse(String args) throws ParseException {
 
         String commandInput = argMultimap.getPreamble();
 
-        Index fromIndex;
-        Index toIndex;
+        Index fromIndex = Index.fromZeroBased(0);
+        Index toIndex = Index.fromZeroBased(0);
+
+        if (argMultimap.getValue(PREFIX_RANGE_FROM).isEmpty() || argMultimap.getValue(PREFIX_RANGE_TO).isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    RangeCommand.MESSAGE_USAGE));
+        }
 
         if (argMultimap.getValue(PREFIX_RANGE_FROM).isPresent() && argMultimap.getValue(PREFIX_RANGE_TO).isPresent()) {
```

Comment:
Is this a redundant check now that you have the guard clause from above?

Reply:
the above guard is it requireNonNull(args)? it only conatin entire args